### PR TITLE
Directories 7

### PR DESCRIPTION
**Pull request by soc**
- Split `BaseDirectories` into `BaseDirectories` (cache, config, ...)
  and `UserDirectories` (audio, video, ...)
- Fields of `BaseDirectories` and `UserDirectories` are now public
- Add substantial amounts of documentation
- The fields of `UserDirectories` are now `null` if no values can be
  retrieved from `xdg-user-dirs`, instead of falling back on hard-coded
  values
- Changes to some values of `BaseDirectories` and `ProjectDirectories`

### CODE REVIEW

File: src/main/java/io/github/soc/directories/ProjectDirectories.java [hunk 1]
```diff
@@ -42,68 +50,88 @@ public static ProjectDirectories fromPath(String path) {
...
-        cacheDir     = homeDir + "/Library/Caches/"              + path_;
-        configDir    = homeDir + "/Library/Preferences/"         + path_;
-        dataDir      = homeDir + "/Library/Application Support/" + path_;
+        homeDir      = System.getProperty("user.home");
+        cacheDir     = homeDir + "/Library/Caches/"              + path;
+        configDir    = homeDir + "/Library/Preferences/"         + path;
+        dataDir      = homeDir + "/Library/Application Support/" + path;
         dataLocalDir = dataDir;
         break;
       case WIN:
```
Comment (soc) [inline src/main/java/io/github/soc/directories/ProjectDirectories.java]: Do the following changes cause any breakage for you, @alexarchambault and @wisechengyi?

Comment (alexarchambault) [reply src/main/java/io/github/soc/directories/ProjectDirectories.java]: It should be ok, we're breaking binary compatiblity with coursier 1.0.x in the current master (1.1.x).

Comment (soc) [reply src/main/java/io/github/soc/directories/ProjectDirectories.java]: Thanks! Looking at your last changes, you are only using `coursierDirs.cacheDir`, not `coursierDirs.configDir` or `coursierDirs.dataDir`, right?

If this is the case, I think no changes are required. What happened is that all paths of `ProjectDirectories` have sub-directories now (like cacheDir already had), increasing consistency, and hopefully reducing the potential of clashes for people with weird Windows setups.

Comment (soc) [reply src/main/java/io/github/soc/directories/ProjectDirectories.java]: @alexarchambault Weird, it shows the changes above the lines I was asking about ... to be precise I'm interested in your opinion on lines 60 (old) to 73 (new). This should be the only thing that might be relevant to you.

**Pull request merged**